### ISSUE TITLE
Unable to load dynamic library 'pdo_sqlsrv.so'/'sqlsrv.so' - No such file or directory - Debian 10.9

### ISSUE THREAD

**Issue by JabWik** [microsoft/msphpsql#1255]
Hello, I have this error all the time:
`PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_sqlsrv.so' (tried: /usr/lib/php/20190902/pdo_sqlsrv.so (/usr/lib/php/20190902/pdo_sqlsrv.so: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/pdo_sqlsrv.so.so (/usr/lib/php/20190902/pdo_sqlsrv.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0`

and

`PHP Warning:  PHP Startup: Unable to load dynamic library 'sqlsrv.so' (tried: /usr/lib/php/20190902/sqlsrv.so (/usr/lib/php/20190902/sqlsrv.so: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/sqlsrv.so.so (/usr/lib/php/20190902/sqlsrv.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0`

I checked my files and it is true that they are not in `20190902`, but they are in another directory, `20200930`. I wrote down the direct path and changed my config file:

`extension=sqlsrv.so -> extension=/usr/lib/php/20200930/sqlsrv.so`
`extension=pdo_sqlsrv.so -> extension=/usr/lib/php/20200930/pdo_sqlsrv.so`

but it did nothing, my error just get different:

`PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_sqlsrv.so' (tried: /usr/lib/php/20200930/pdo_sqlsrv.so (/usr/lib/php/20200930/pdo_sqlsrv.so: cannot open shared object file: No such file or directory), /usr/lib/php/20200930/pdo_sqlsrv.so.so (/usr/lib/php/20200930/pdo_sqlsrv.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0`

`PHP Warning:  PHP Startup: Unable to load dynamic library 'sqlsrv.so' (tried: /usr/lib/php/20200930/sqlsrv.so (/usr/lib/php/20200930/sqlsrv.so: cannot open shared object file: No such file or directory), /usr/lib/php/20200930/sqlsrv.so.so (/usr/lib/php/20200930/sqlsrv.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0`

I installed everything step-by-step from the official tutorial for Linux Debian: https://docs.microsoft.com/en-us/sql/connect/php/installation-tutorial-linux-mac?view=sql-server-ver15#installing-on-debian I tried to reinstall the drivers again and again, but every time they get them installed to the another directory and somehow PHP does not see them even if I specify the direct path for them.

I tried different solutions like editing the php.ini file and manually loading the modues, modifying the pdo.ini / sqlsrv.ini / pdo_sqlsrv.ini files, I checked their loading order. I even tried to copy files from `20190902` to `20200930`. I remembered to restart the web server everytime when I changed anything, just to be sure that the changes that had been made, were loaded. Honestly, I'm out of ideas what I can do more. My extensions are still not working.

My php.ini:
Configuration File (php.ini) Path: /etc/php/7.4/cli
Loaded Configuration File:         /etc/php/7.4/cli/php.ini
Scan for additional .ini files in: /etc/php/7.4/cli/conf.d
Additional .ini files parsed:      /etc/php/7.4/cli/conf.d/10-mysqlnd.ini,
/etc/php/7.4/cli/conf.d/10-opcache.ini,
/etc/php/7.4/cli/conf.d/10-pdo.ini,
/etc/php/7.4/cli/conf.d/15-xml.ini,
/etc/php/7.4/cli/conf.d/20-calendar.ini,
/etc/php/7.4/cli/conf.d/20-ctype.ini,
/etc/php/7.4/cli/conf.d/20-dom.ini,
/etc/php/7.4/cli/conf.d/20-exif.ini,
/etc/php/7.4/cli/conf.d/20-ffi.ini,
/etc/php/7.4/cli/conf.d/20-fileinfo.ini,
/etc/php/7.4/cli/conf.d/20-ftp.ini,
/etc/php/7.4/cli/conf.d/20-gettext.ini,
/etc/php/7.4/cli/conf.d/20-iconv.ini,
/etc/php/7.4/cli/conf.d/20-json.ini,
/etc/php/7.4/cli/conf.d/20-mongodb.ini,
/etc/php/7.4/cli/conf.d/20-mysqli.ini,
/etc/php/7.4/cli/conf.d/20-pdo_mysql.ini,
/etc/php/7.4/cli/conf.d/20-pdo_sqlsrv.ini,
/etc/php/7.4/cli/conf.d/20-phar.ini,
/etc/php/7.4/cli/conf.d/20-posix.ini,
/etc/php/7.4/cli/conf.d/20-readline.ini,
/etc/php/7.4/cli/conf.d/20-shmop.ini,
/etc/php/7.4/cli/conf.d/20-simplexml.ini,
/etc/php/7.4/cli/conf.d/20-sockets.ini,
/etc/php/7.4/cli/conf.d/20-sqlsrv.ini,
/etc/php/7.4/cli/conf.d/20-sysvmsg.ini,
/etc/php/7.4/cli/conf.d/20-sysvsem.ini,
/etc/php/7.4/cli/conf.d/20-sysvshm.ini,
/etc/php/7.4/cli/conf.d/20-tokenizer.ini,
/etc/php/7.4/cli/conf.d/20-xmlreader.ini,
/etc/php/7.4/cli/conf.d/20-xmlwriter.ini,
/etc/php/7.4/cli/conf.d/20-xsl.ini,
/etc/php/7.4/cli/conf.d/30-pdo_sqlsrv.ini

PHP version: 7.4.18 NTS
Linux: Debian 10.9

The other modules are working fine, just having problems with sqlsrv and pdo_sqlsrv. Thank you in advance for your help, I can provide you any additional information if you need any.

**Comment by yitam** [comment]
Hi @JabWik 

Judging from the include path, as explained in [this FAQ entry](https://github.com/microsoft/msphpsql/wiki/FAQ#recipe-for-target-failed), I think you've installed the sqlsrv drivers for PHP 8.0 instead of 7.4. Please check [the instructions](https://docs.microsoft.com/sql/connect/php/installation-tutorial-linux-mac?view=sql-server-ver15#installing-on-debian) again and replace 8.0 with 7.4

Is it intentional to keep multiple PHP versions?

**Comment by JabWik** [comment]
Hello, thank you for your reply!

To be honest, I have no idea why we had PHP 8.0 on our web server. We do not use it in any of our projects - so first of all, I decided to delete it. My current package list:
`libapache2-mod-php7.4      
php-common  
php-pear  
php7.0-common 
php7.0-xml  
php7.4    
php7.4-cli      
php7.4-common      
php7.4-dev        
php7.4-json   
php7.4-mongodb        
php7.4-mysql             
php7.4-opcache
php7.4-readline    
php7.4-xml   
pkg-php-tools`

I installed the extension according to the instructions and this time the packages landed in the appropriate folder. I am just suprised, because my commands were the same as before the weekend. You know, I was very carefull to put "7.4" where it was needed and so on.

`Build process completed successfully
Installing '/usr/lib/php/20190902/sqlsrv.so'
install ok: channel://pecl.php.net/sqlsrv-5.9.0`
`Build process completed successfully
Installing '/usr/lib/php/20190902/pdo_sqlsrv.so'
install ok: channel://pecl.php.net/pdo_sqlsrv-5.9.0`

And finally, it works! The errors are gone and the extensions are visible on the list:
` [PHP Modules]
...
pdo_sqlsrv
...
sqlsrv
...
`

I just have this warning, where can I look for it to repair?
`PHP Warning:  Module 'pdo_sqlsrv' already loaded in Unknown on line 0`

I tested my connection with the PHP file provided from: https://docs.microsoft.com/pl-pl/sql/connect/php/installation-tutorial-linux-mac?view=sql-server-ver15#testing-your-installation 
Unfortunately, I can not establish the connection with the server, so that is another problem to solve. However, I am very glad that the extensions are working at long last! :sparkling_heart:

**Comment by JabWik** [comment]
Yes, I have installed the ODBC driver on our machine:
`msodbcsql17`
`mssql-tools`

When I try to connect with my database from Linux Debian, I have such errors:
`SQL Error:
Error information:
SQLSTATE: 08001
Code: 10054
Message: [Microsoft][ODBC Driver 17 for SQL Server]TCP Provider: Error code 0x2746
SQLSTATE: 08001
Code: 10054
Message: [Microsoft][ODBC Driver 17 for SQL Server]Client unable to establish connection`

I checked my credenstials. I am able to connect to the base using other programs and the base is opened for remote connections.

However, the point is that we have another machine with Windows Server 2019 and everything works fine on Windows. We use the same PHP version, modules and drivers. We even use the same connection method.
The machine with Windows is able to connect, the machine with Linux is not able to connect.

At the moment all data flows through the Windows machine to Linux machine,  but this is a very ineffective solution. 

The version of the data base I try to connect: SQL Server 2012.
